File: code/java/ch15/practice/Practice04SubarraySumEqualsK.java
package ch15.practice;

import java.util.*;

/**
 * Practice 4: Subarray Sum Equals K (Sliding Window)
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 *
 * PROBLEM: Count subarrays with sum == k (positive integers only).
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Practice04SubarraySumEqualsK {
    public static int solve(int[] arr, int k) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine().trim();
        int[] arr = Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
        int k = sc.nextInt();
        System.out.println(solve(arr, k));
        sc.close();
    }
}

File: code/python/ch15/practice/practice_04_subarray_sum_equals_k.py
"""
Practice 4: Subarray Sum Equals K (Sliding Window)
====================================================
Chapter 15: Two Pointers & Sliding Window — The Dance of Indices

PROBLEM
-------
Given an array of positive integers and a target sum k, count the number
of contiguous subarrays whose sum equals exactly k.

INPUT FORMAT
------------
First line: space-separated positive integers.
Second line: a single integer k.

OUTPUT FORMAT
-------------
A single integer — the count of subarrays with sum equal to k.

CONSTRAINTS
-----------
- 1 <= len(arr) <= 10^5
- 1 <= arr[i] <= 10^4
- 1 <= k <= 10^9

EXAMPLES
--------
Input:
  1 1 1
  2
Output: 2

Input:
  1 2 3
  3
Output: 2

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(arr: list[int], k: int) -> int:
    """Return count of subarrays with sum equal to k."""
    pass  # TODO: Replace this with your solution



# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    line = input().strip()
    arr = list(map(int, line.split()))
    k = int(input().strip())
    print(solve(arr, k))

